db/migrate/20160324203119_add_id_primary_key_to_join_tables.rb

```diff
@@ -0,0 +1,82 @@
+require_relative './20160106214719_add_composite_primary_keys_to_join_tables'
+
+class AddIdPrimaryKeyToJoinTables < ActiveRecord::Migration[5.0]
+  JOIN_TABLES = %w(
+    conditions_miq_policies
+    key_pairs_vms
+    miq_roles_features
+    miq_servers_product_updates
+    security_groups_vms
+    storages_vms_and_templates
+    miq_groups_users
+    cloud_tenants_vms
+    customization_scripts_operating_system_flavors
+    configuration_locations_configuration_profiles
+    configuration_organizations_configuration_profiles
+    configuration_profiles_configuration_tags
+    configuration_tags_configured_systems
+    container_groups_container_services
+    direct_configuration_profiles_configuration_tags
+    direct_configuration_tags_configured_systems
+    network_ports_security_groups
+  ).freeze
+
+  class MiqRegion < ApplicationRecord
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    say_with_time("Removing composite primary keys from join tables") do
+      revert AddCompositePrimaryKeysToJoinTables
+    end
+
+    JOIN_TABLES.each do |t|
+      delete_remote_region_rows(t) if on_replication_target?
+
+      say_with_time("Add primary key \"id\" to #{t}") do
+        execute <<-SQL
+          CREATE SEQUENCE #{sequence_name(t)} START #{seq_start_value}
+        SQL
+
+        execute <<-SQL
+          ALTER TABLE #{t} ADD COLUMN id BIGINT PRIMARY KEY
+          DEFAULT NEXTVAL('#{sequence_name(t)}')
+        SQL
+
+        execute <<-SQL
+          ALTER SEQUENCE #{sequence_name(t)} OWNED BY #{t}.id
+        SQL
+      end
+    end
+  end
+
+  def down
+    JOIN_TABLES.each do |t|
+      remove_column t.to_sym, :id
+    end
+
+    run AddCompositePrimaryKeysToJoinTables
+  end
+
+  def delete_remote_region_rows(table)
+    execute("SELECT * FROM #{table}").each do |row|
+      next if ApplicationRecord.id_in_current_region?(row.values.first)
```

Comment:
@Fryguy I can understand wanting to simplify this if it were in application code. But, since it's in a migration and will be executed one time and -shouldn't- need to be maintained, @carbonin and I were thinking the best approach was to do it with least amount of code. Simplifying the query at the expense of adding definitions for each of the affected models seems to go against that. Maybe we should have a quick discussion about so that we can proceed with getting this PR merged,